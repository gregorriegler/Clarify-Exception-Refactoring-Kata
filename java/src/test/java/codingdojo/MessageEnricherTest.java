package codingdojo;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;


public class MessageEnricherTest {

    @Test
    void sampleTest() {
        MessageEnricher enricher = new MessageEnricher();
        SpreadsheetWorkbook worksheet = new SpreadsheetWorkbook();
        Exception e = new RuntimeException("Terrible problem");

        ErrorResult actual = enricher.enrichError(worksheet, e);

        Approvals.verify(actual);
    }

    @Test
    void expressionParseException() {
        MessageEnricher enricher = new MessageEnricher();
        SpreadsheetWorkbook worksheet = new SpreadsheetWorkbook();
        Exception e = new ExpressionParseException("Terrible problem");

        ErrorResult actual = enricher.enrichError(worksheet, e);

        Approvals.verify(actual);
    }

    @Test
    void circularReferenceException() {
        MessageEnricher enricher = new MessageEnricher();
        SpreadsheetWorkbook worksheet = new SpreadsheetWorkbook();
        Exception e = new RuntimeException("Circular Reference And something else");

        ErrorResult actual = enricher.enrichError(worksheet, e);

        Approvals.verify(actual);
    }

    @Test
    void vLookupException() {
        MessageEnricher enricher = new MessageEnricher();
        SpreadsheetWorkbook worksheet = new SpreadsheetWorkbook();
        Exception e = irrelevant_vLookup_irrelevant();

        ErrorResult actual = enricher.enrichError(worksheet, e);

        Approvals.verify(actual);
    }

    @Test
    void noMatchesFoundException() {
        MessageEnricher enricher = new MessageEnricher();
        SpreadsheetWorkbook worksheet = new SpreadsheetWorkbook();
        Exception e = new RuntimeException("No matches found");

        ErrorResult actual = enricher.enrichError(worksheet, e);

        Approvals.verify(actual);
    }

    private RuntimeException irrelevant_vLookup_irrelevant() {
        RuntimeException exception = new RuntimeException("Object reference not set to an instance of an object");
        exception.fillInStackTrace();
        return exception;
    }

}
