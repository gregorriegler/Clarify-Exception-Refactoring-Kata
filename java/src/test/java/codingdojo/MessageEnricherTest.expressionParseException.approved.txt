ErrorResult{formulaName='Name of the formula', message='Invalid expression found in tax formula [Name of the formula]. Check that separators and delimiters use the English locale.', presentation='Presentation of which cells in the spreadsheet are populated with numbers and formulas and things'}